sdks/java/core/src/main/java/org/apache/beam/sdk/util/IOChannelUtils.java

```diff
@@ -202,4 +202,25 @@ public static IOChannelFactory getFactory(String spec) throws IOException {
   public static String resolve(String path, String other) throws IOException {
     return getFactory(path).resolve(path, other);
   }
+
+  /**
+   * Resolve given multiple {@code others} against the {@code path}.
+   *
+   * <p>If the {@code others} parameter contains empty paths then empty paths are ignored. If the
```

Comment:
<p>Empty paths in {@code others} are ignored. If {@code others} contains one or more absolute paths, then this method returns a path that starts with the last absolute path in {@code others} joined with the remaining paths. Resolution of paths is highly implementation dependent and therefore unspecified.